packages/did/src/Did.chain.ts

```diff
@@ -387,21 +391,20 @@ interface IDidAuthorizedCallOperation extends Struct {
 
 // ### EXTRINSICS
 
-type FormattedPublicKey<K extends NewDidKey> = TypedValue<
-  Capitalize<K['type']>,
-  K['publicKey']
->
+export function encodePublicKey(
+  key: NewDidVerificationKey
+): EncodedVerificationKey
+export function encodePublicKey(key: NewDidEncryptionKey): EncodedEncryptionKey
```

Comment:
These declarations provide the narrowing of types, necessary for some cases

Reply:
I do like the idea of using function overloads to deal with key type subsets, but we will have other subsets too. For example in the account linking pallet we will be be dealing with an additional signature type soon: `Ethereum`